### PR TITLE
add linkNonce/linkID input/output

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: circuits/offchain/credentialAtomicQueryV3OffChain.circom [hunk 1]
```diff
@@ -2,11 +2,13 @@ pragma circom 2.1.5; /* chage to 2.1.6 */
 
 include "./credentialAtomicQueryMTPOffChain.circom";
 include "./credentialAtomicQuerySigOffChain.circom";
+include "../lib/linked/linkId.circom";
 
 template credentialAtomicQueryV3OffChain(issuerLevels, claimLevels, valueArraySize) {
     // common outputs for Sig and MTP
     signal output merklized;
     signal output userID;
+    signal output linkID;
```
Comment (OBrezhniev) [inline circuits/offchain/credentialAtomicQueryV3OffChain.circom:11]: Please move it both linkID and linkNonce to the end of signals block (after issuerAuthState)

File: circuits/onchain/credentialAtomicQueryV3OnChain.circom [hunk 2]
```diff
@@ -45,6 +45,8 @@ template credentialAtomicQueryV3OnChain(issuerLevels, claimLevels, valueArraySiz
     // unless nonce == 0, in which case userID will be assigned with userGenesisID
     signal output userID;
 
+    signal output linkID;
```
Comment (OBrezhniev) [inline circuits/onchain/credentialAtomicQueryV3OnChain.circom:48]: Same here.

**Review by OBrezhniev — CHANGES_REQUESTED** [review]
Please add testvector generators and unit tests for non-zero linkNonce.

File: testvectorgen/utils/utils.go [hunk 3]
```diff
@@ -256,3 +257,31 @@ func PoseidonHashValue(values []*big.Int) (*big.Int, error) {
...
 	return fullHash, nil
 }
+
+// CalculateLinkID returns linkID calculated from linkNonce and claim
+func CalculateLinkID(linkNonce string, claim *core.Claim) (string, error) {
+	if linkNonce == "0" {
+		return "0", nil
+	}
+
+	nonceInt, err := strconv.Atoi(linkNonce)
```
Comment (OBrezhniev) [inline testvectorgen/utils/utils.go:267]: linkNonce can be 254-bit integer, so convert to big.Int directly.

**Review by OBrezhniev — CHANGES_REQUESTED** [review]

**Review by OBrezhniev — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]
